utils/templates/types.py.tpl

```diff
@@ -0,0 +1,84 @@
+#  Licensed to Elasticsearch B.V. under one or more contributor
+#  license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright
+#  ownership. Elasticsearch B.V. licenses this file to you under
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+from typing import Any, Dict, Literal, Mapping, Sequence, Union
+
+from elastic_transport.client_utils import DEFAULT, DefaultType
+
+from elasticsearch_dsl.document_base import InstrumentedField
+from elasticsearch_dsl import function, Query
+from elasticsearch_dsl.utils import AttrDict
+
+PipeSeparatedFlags = str
+
+
+{% for k in classes %}
+class {{ k.name }}({% if k.parent %}{{ k.parent }}{% else %}AttrDict[Any]{% endif %}):
```

Comment:
I believe using an [inline if-expression](https://jinja.palletsprojects.com/en/3.1.x/templates/#if-expression) would be clearer:

```suggestion
class {{ k.name }}({ k.parent if k.parent else "AttrDict[Any]" }}):
```

Reply:
I didn't know you could use inline if statements in Jinja. Thanks.